### ISSUE TITLE
Wakelocks and Alarms are always 0

### ISSUE THREAD

**Issue by gregferns** [rsteckler/unbounce-android#121]
Hi. I've tried this app on so many different roms, CM based, AOSP based, etc etc on my YUPHORIA (lettuce)(twrp recovery). All these roms are 5.0.x and above,, even 5.1.1... Never got to try Kitkat as there isn't any rom for my phone for that...

Running latest xposed framework with alpha4... Had even tried previous versions with alpha 3 also but to no effect... Always the same problem. No wakelocks or alarms are blocked... using donate version so all wakelocks and alarms can be set and ARE SET,, but no results.

Xposed works fine for other modules and this one also is properly installed but i always see in xposed logcat the line AMPLIFY STAT FILE RESET UPON VERSION UPGRADE.... Dont know if anything wrong with that...

Can't post log here so giving link please
https://www.dropbox.com/s/7163rg24adp2xtg/xposed_error_20150805_115603.log?dl=0

Edit : have tried with other modules also like greenify, powernap and also tried without any modules, only amplify and xposed framework... same results...

Please try and solve this as i want to use this app....

thanks in advance
Greg

**Comment by v-yadli** [comment]
Same issue here with ZTE nubia Z9 on stock 5.0.2. I'm also on xposed 3.0 alpha 4.

Here's some logcat coming from the app:

```
logcat | grep com.ryansteckler
I/ActivityManager( 1134): Killing 6912:com.ryansteckler.nlpunbounce/u0a132 (adj 15): empty #17
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.hooks.Wakelocks.resetFilesIfNeeded:93 com.ryansteckler.nlpunbounce.hooks.Wakelocks.updateStatsIfNeeded:514 com.ryansteckler.nlpunbounce.hooks.Wakelocks.handleAlarm:537 com.ryansteckler.nlpunbounce.hooks.Wakelocks.access$300:40
I/ActivityManager( 1134): Start proc com.ryansteckler.nlpunbounce for broadcast com.ryansteckler.nlpunbounce/.ActivityReceiver: pid=8850 uid=10132 gids={50132, 9997, 3003} abi=arm64-v8a
I/ActivityManager( 1134): START u0 {dat=file:///data/data/com.ryansteckler.nlpunbounce/shared_prefs/com.ryansteckler.nlpunbounce_preferences.xml cmp=com.speedsoftware.rootexplorer/.DisplayText (has extras)} from uid 10087 on display 0
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.hooks.Wakelocks.resetFilesIfNeeded:93 com.ryansteckler.nlpunbounce.hooks.Wakelocks.updateStatsIfNeeded:514 com.ryansteckler.nlpunbounce.hooks.Wakelocks.handleAlarm:537 com.ryansteckler.nlpunbounce.hooks.Wakelocks.access$300:40
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.hooks.Wakelocks.resetFilesIfNeeded:93 com.ryansteckler.nlpunbounce.hooks.Wakelocks.updateStatsIfNeeded:514 com.ryansteckler.nlpunbounce.hooks.Wakelocks.handleAlarm:537 com.ryansteckler.nlpunbounce.hooks.Wakelocks.access$300:40
I/Timeline( 2268): Timeline: Activity_launch_request id:com.ryansteckler.nlpunbounce time:1315610
I/ActivityManager( 1134): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.ryansteckler.nlpunbounce/.Settings-Alias bnds=[540,1284][798,1560] (has extras)} from uid 1000 on display 0
I/jiepp   ( 2500): packageName com.ryansteckler.nlpunbounce
D/Finsky  ( 6539): [480] InAppBillingUtils.getPreferredAccount: com.ryansteckler.nlpunbounce: Account determined from library ownership - [UvgIag9YiD1PyT-arTZMX90A7MM]
D/Finsky  ( 6539): [449] InAppBillingUtils.getPreferredAccount: com.ryansteckler.nlpunbounce: Account determined from library ownership - [UvgIag9YiD1PyT-arTZMX90A7MM]
D/Finsky  ( 6539): [450] InAppBillingUtils.getPreferredAccount: com.ryansteckler.nlpunbounce: Account determined from library ownership - [UvgIag9YiD1PyT-arTZMX90A7MM]
D/Finsky  ( 6539): [480] InAppBillingUtils.getPreferredAccount: com.ryansteckler.nlpunbounce: Account determined from library ownership - [UvgIag9YiD1PyT-arTZMX90A7MM]
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.XposedReceiver.onReceive:43 android.app.LoadedApk$ReceiverDispatcher$Args.run:870 android.os.Handler.handleCallback:739 android.os.Handler.dispatchMessage:95
I/ActivityManager( 1134): Displayed com.ryansteckler.nlpunbounce/.Settings-Alias: +823ms
I/Timeline( 1134): Timeline: Activity_windows_visible id: ActivityRecord{22311ca4 u0 com.ryansteckler.nlpunbounce/.Settings-Alias t376} time:1316473
I/Timeline( 8850): Timeline: Activity_launch_request id:com.ryansteckler.nlpunbounce time:1322750
I/ActivityManager( 1134): START u0 {cmp=com.ryansteckler.nlpunbounce/.SettingsActivity} from uid 10132 on display 0
I/ActivityManager( 1134): Displayed com.ryansteckler.nlpunbounce/.SettingsActivity: +382ms
I/Timeline( 1134): Timeline: Activity_windows_visible id: ActivityRecord{8fbc4ad u0 com.ryansteckler.nlpunbounce/.SettingsActivity t376} time:1323579
I/Timeline( 1134): Timeline: Activity_windows_visible id: ActivityRecord{8fbc4ad u0 com.ryansteckler.nlpunbounce/.SettingsActivity t376} time:1327956
I/WindowManager( 1134): Screen frozen for +395ms due to Window{37ec2748 u0 com.ryansteckler.nlpunbounce/com.ryansteckler.nlpunbounce.SettingsActivity}
I/Timeline( 1134): Timeline: Activity_windows_visible id: ActivityRecord{8fbc4ad u0 com.ryansteckler.nlpunbounce/.SettingsActivity t376} time:1365194
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.hooks.Wakelocks.resetFilesIfNeeded:93 com.ryansteckler.nlpunbounce.hooks.Wakelocks.updateStatsIfNeeded:514 com.ryansteckler.nlpunbounce.hooks.Wakelocks.handleAlarm:537 com.ryansteckler.nlpunbounce.hooks.Wakelocks.access$300:40
I/Timeline( 1134): Timeline: Activity_windows_visible id: ActivityRecord{22311ca4 u0 com.ryansteckler.nlpunbounce/.Settings-Alias t376} time:1911393
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.XposedReceiver.onReceive:43 android.app.LoadedApk$ReceiverDispatcher$Args.run:870 android.os.Handler.handleCallback:739 android.os.Handler.dispatchMessage:95
I/Timeline( 8850): Timeline: Activity_launch_request id:com.ryansteckler.nlpunbounce time:1941428
I/ActivityManager( 1134): START u0 {cmp=com.ryansteckler.nlpunbounce/.SettingsActivity} from uid 10132 on display 0
I/ActivityManager( 1134): Displayed com.ryansteckler.nlpunbounce/.SettingsActivity: +319ms
I/Timeline( 1134): Timeline: Activity_windows_visible id: ActivityRecord{22d03642 u0 com.ryansteckler.nlpunbounce/.SettingsActivity t376} time:1942180
I/Timeline( 1134): Timeline: Activity_windows_visible id: ActivityRecord{22311ca4 u0 com.ryansteckler.nlpunbounce/.Settings-Alias t376} time:1953564
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.hooks.Wakelocks.resetFilesIfNeeded:93 com.ryansteckler.nlpunbounce.hooks.Wakelocks.updateStatsIfNeeded:514 com.ryansteckler.nlpunbounce.hooks.Wakelocks.handleAlarm:537 com.ryansteckler.nlpunbounce.hooks.Wakelocks.access$300:40
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.XposedReceiver.onReceive:43 android.app.LoadedApk$ReceiverDispatcher$Args.run:870 android.os.Handler.handleCallback:739 android.os.Handler.dispatchMessage:95
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.XposedReceiver.onReceive:43 android.app.LoadedApk$ReceiverDispatcher$Args.run:870 android.os.Handler.handleCallback:739 android.os.Handler.dispatchMessage:95
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.hooks.Wakelocks.resetFilesIfNeeded:93 com.ryansteckler.nlpunbounce.hooks.Wakelocks.updateStatsIfNeeded:514 com.ryansteckler.nlpunbounce.hooks.Wakelocks.handleAlarm:537 com.ryansteckler.nlpunbounce.hooks.Wakelocks.access$300:40
I/Timeline( 2268): Timeline: Activity_launch_request id:com.ryansteckler.nlpunbounce time:2556257
I/ActivityManager( 1134): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.ryansteckler.nlpunbounce/.Settings-Alias bnds=[540,1284][798,1560] (has extras)} from uid 1000 on display 0
I/jiepp   ( 2500): packageName com.ryansteckler.nlpunbounce
I/Timeline( 1134): Timeline: Activity_windows_visible id: ActivityRecord{22311ca4 u0 com.ryansteckler.nlpunbounce/.Settings-Alias t376} time:2556944
W/ContextImpl( 1134): Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:1416 com.ryansteckler.nlpunbounce.hooks.Wakelocks.resetFilesIfNeeded:93 com.ryansteckler.nlpunbounce.hooks.Wakelocks.updateStatsIfNeeded:514 com.ryansteckler.nlpunbounce.hooks.Wakelocks.handleAlarm:537 com.ryansteckler.nlpunbounce.hooks.Wakelocks.access$300:40
```

**Comment by v-yadli** [comment]
update: I've tracked down the issue. It seems to be XSharedPreferences.
When I add:
            defaultLog("Wakelock " + wakeLockName + ": prefs indicate that we should ignore.");
to handleWakeLockAcquire, I see that all wakelocks are ignored. No wonder that they will show as zero.

@rsteckler The google location stuff is draining battery too much for me. I'm going to hard code the names and use my modified version until there's a fix from upstream...

I should've noticed https://github.com/rovo89/XposedBridge/issues/58 earlier!

**Comment by rsteckler** [comment]
Hi all,
This was caused by a bug in Xposed on 64 bit architectures.  It was fixed in Xposed by  rovo89/Xposed@46cedbc .  

Nothing needed in Amplify.
